_posts/en/newsletters/2024-06-14-newsletter.md

```diff
@@ -0,0 +1,149 @@
+startup time when a user restarts their node before an ongoing reindex
+has completed.
+
+Bitcoin Core implements five indexes. The UTXO set and the block index
+are required, whereas the transaction index, compact block filter index
+and coinstats index are optional. When running with `-reindex`, all
+indexes are wiped and rebuilt. This process can take quite a while, and
+it is not guaranteed to finish before the node is stopped for any
+reason.
+
+Because the node needs to have an up to date UTXO set and block index,
+the reindexing status is persisted on disk. When a reindex is started, a
+flag is [set][reindex flag set], and it will only be unset when the
+reindex is finished. This way, when the node starts, it can detect that
+it should continue reindexing, even if the user didn’t provide the flag
+as a startup option.
+
+When restarting (without `-reindex`) after an unfinished reindex, the
+required indexes are preserved and will continue to be rebuilt further.
+Prior to [Bitcoin Core #30132][], the optional indexes would be wiped a
+second time. [Bitcoin Core #30132][] can make node startup more
+efficient by avoiding the wiping of the optional indexes when it is not
+necessary.
+
+{% include functions/details-list.md
+  q0="What is the behaviour change introduced by this PR?"
+  a0="Behaviour is changed in three ways. First, as per the goal of this
+  PR, optional indexes are no longer wiped again when the node is
+  restarted before reindexing is completed. This aligns the wiping
+  behaviour of optional indexes with that of required indexes. Second,
+  when a user requests a reindex through the GUI, this request is no
+  longer ignored, reversing a subtle bug introduced in [b47bd95][gh
+  b47bd95]. Third, the log line \"Initializing databases...\\n\" is
+  removed."
+  a0link="https://bitcoincore.reviews/30132#l-19"
+
+  q1="What are the two ways an optional index can process new blocks?"
+  a1="When an optional index is initialized, it checks if its highest
+  block is the same as the current chaintip. If it is not, it will first
+  process all missing blocks with a background sync, through
+  `BaseIndex::StartBackgroundSync()`. As soon as the index catches up
+  with the chaintip, it will process all further blocks through the
+  validation interface using `ValidationSignals::BlockConnected`."
+  a1link="https://bitcoincore.reviews/30132#l-52"
+
+  q2="How does this PR affect the logic of optional indexes processing
+  new blocks?"
+  a2="Prior to this PR, wiping the optional indexes without wiping the
+  chainstate means these indexes will be considered out-of-sync. As per
+  the previous question, that means they will first perform a background
+  sync before switching to the validation interface. With this PR, the
+  optional indexes remain in sync with the chainstate, and as such no
+  background sync is required. Note: background sync only starts after
+  reindex has completed, whereas processing validation events happens in
```

Comment:
```suggestion
  reindex has completed, whereas processing validation events happen in
```

Reply:
I think "happens" is correct here, so I've left that as is but I've taken all your other suggestions, thank you!